feature-libs/customer-ticketing/components/customer-ticketing-messages/customer-ticketing-messages.component.ts

```diff
@@ -1,112 +1,110 @@
-          },
-        ],
-      },
-      {
-        author: 'Mark Rivers',
-        createdAt: '2022-06-22T14:37:15+0000',
-        message: 'A message to consider',
-      },
-      {
-        addedByAgent: true,
-        createdAt: '2022-06-22T20:25:02+0000',
-        message: 'This is the way',
-      },
-      {
-        addedByAgent: true,
-        createdAt: '2022-06-22T20:25:02+0000',
-        message: 'This is the way',
-      },
-      {
-        author: 'Mark Rivers',
-        createdAt: '2022-06-22T14:37:15+0000',
-        message: 'A message to consider',
-      },
-    ],
-  });
+export class CustomerTicketingMessagesComponent implements OnDestroy {
+  ticketDetails$: Observable<TicketDetails | undefined> =
+    this.customerTicketingFacade.getTicket();
 
-  constructor(protected customerTicketingConfig: CustomerTicketingConfig) {}
+  constructor(
+    protected customerTicketingConfig: CustomerTicketingConfig,
+    protected customerTicketingFacade: CustomerTicketingFacade,
+    protected eventService: EventService
+  ) {}
+
+  subscription = new Subscription();
 
   messageEvents$: Observable<Array<MessageEvent> | undefined> =
     this.prepareMessageEvents();
 
   messagingConfigs: MessagingConfigs = this.prepareMessagingConfigs();
 
-  onSend(_event: { files: FileList | undefined; message: string }) {
-    // call to submit new event and upload attachment
+  onSend(event: { files: FileList | undefined; message: string }) {
+    this.subscription.add(
+      this.customerTicketingFacade
+        .createTicketEvent(this.prepareTicketEvent(event.message))
+        .subscribe((createdEvent: TicketEvent) => {
+          if (event.files?.length && createdEvent.code) {
+            this.customerTicketingFacade.uploadAttachment(
+              event.files.item(0),
+              createdEvent.code
+            );
+          } else {
+            this.eventService.dispatch({ status }, TicketEventCreatedEvent);
+          }
+        })
+    );
+  }
+
+  downloadAttachment(event: {
+    messageCode: string;
+    attachmentId: string;
+    fileName: string;
+  }) {
+    this.subscription.add(
+      this.customerTicketingFacade
+        .downloadAttachment(event.messageCode, event.attachmentId)
+        .subscribe((data) => {
+          const downloadURL = window.URL.createObjectURL(data as any);
+          const link = document.createElement('a');
+          link.href = downloadURL;
+          link.download = event.fileName;
+          link.click();
+        })
+    );
   }
 
-  prepareMessageEvents(): Observable<Array<MessageEvent> | undefined> {
+  protected prepareMessageEvents(): Observable<
+    Array<MessageEvent> | undefined
+  > {
     return this.ticketDetails$.pipe(
       map((ticket) =>
-        ticket.ticketEvents?.map(
-          (event): MessageEvent => ({
+        ticket?.ticketEvents?.reverse().map(
```

Comment:
How about sorting the list by creation time,  because no matter the order in which the events arrive, you need to ensure that they are always presented in ascending order

Reply:
It is already sorted from the backend, but it is in the reversed order. So, I am reversing it again.

Reply:
Please try adding the sorting in the adapter.